src/components/dialogs/AddTeacher.tsx

```diff
@@ -0,0 +1,85 @@
+// Modules
+import { useRouter } from "next/router";
+import { useTranslation } from "next-i18next";
+import { useEffect, useState } from "react";
+
+// SK Components
+import { Dialog, DialogSection, Dropdown } from "@suankularb-components/react";
+
+// Types
+import { DialogProps } from "@utils/types/common";
+import { Teacher } from "@utils/types/person";
+
+// Helpers
+import { nameJoiner } from "@utils/helpers/name";
+
+// Hooks
+import { useSubjectGroupOption } from "@utils/hooks/subject";
+import { useTeacherOption } from "@utils/hooks/teacher";
+
+const AddTeacherDialog = ({
+  show,
+  onClose,
+  onSubmit,
+}: DialogProps & { onSubmit: (teacher: Teacher) => void }): JSX.Element => {
+  const { t } = useTranslation("common");
+  const locale = useRouter().locale == "en-US" ? "en-US" : "th";
+
+  const subjectGroups = useSubjectGroupOption();
+  const [selectedGroup, setSelectedGroup] = useState<number>(1);
+
+  const teachers = useTeacherOption(selectedGroup);
+  const [selectedTeacher, setSelectedTeacher] = useState<number | null>(
+    teachers.length > 0 ? teachers[0].id : null
+  );
+
+  useEffect(
+    () => setSelectedTeacher(teachers.length > 0 ? teachers[0].id : null),
+    [selectedGroup]
```

Comment:
This might need to depend on `teachers` as well since it require teacher to finish fetching